src/compiler/checker.ts

```diff
@@ -2945,7 +2945,7 @@ namespace ts {
                         // Always use 'typeof T' for type of class, enum, and module objects
                         if (symbol.flags & SymbolFlags.Class && !getBaseTypeVariableOfClass(symbol) && !(symbol.valueDeclaration.kind === SyntaxKind.ClassExpression && context.flags & NodeBuilderFlags.WriteClassExpressionAsTypeLiteral) ||
                             symbol.flags & (SymbolFlags.Enum | SymbolFlags.ValueModule) ||
-                            shouldWriteTypeOfFunctionSymbol()) {
+                            shouldWriteTypeOfFunctionSymbol() && !(context.flags & NodeBuilderFlags.UseStructuralFallback && isSymbolAccessible(symbol, context.enclosingDeclaration, SymbolFlags.Value, /*computeAliases*/ false).accessibility !== SymbolAccessibility.Accessible)) {

```

Comment:
A helper similar to `isTypeSymbolAccessible` would be nice to avoid writing `.accessibility !== SymbolAccessibility.Accessible)`
Also, why not just put the `&&` part inside of `shouldWriteTypeOfFunctionSymbol`?

Reply:
Good point. I'll move it into `shouldWriteTypeOfFunctionSymbol`.